netplay/createroom.cpp

```diff
@@ -148,7 +157,17 @@ void CreateRoom::handleCreateButton()
         webSocket = new QWebSocket;
         (*CoreDoCommand)(M64CMD_ROM_GET_SETTINGS, sizeof(rom_settings), &rom_settings);
         connect(webSocket, &QWebSocket::connected, this, &CreateRoom::onConnected);
-        webSocket->open(QUrl(serverChooser->currentData().toString()));
+        connect(webSocket, SIGNAL(error(QAbstractSocket::SocketError)), this, SLOT(handleConnectionError(QAbstractSocket::SocketError)));
+        QString serverAddress = serverChooser->currentData() == "Custom" ? customServerHost.prepend("ws://") : serverChooser->currentData().toString();
+        QUrl serverUrl = QUrl(serverAddress);
+        if (serverChooser->currentData() == "Custom" && serverUrl.port() < 0)
+            // Be forgiving of custom server addresses that forget the port
+            serverUrl.setPort(45000);
+        connectionTimer = new QTimer(this);
```

Comment:
This timer should not be needed. The webSocket's error signal will emit if the connection doesn't succeed, and you've already got that connected to ```handleConnectionError```

Reply:
In my testing, the error signal wasn't fired when I typed in a wrong IP address. I don't know why.

Reply:
OK, it's possible it doesn't count as an error I guess. You might want to try the stateChanged signal: https://doc.qt.io/qt-5/qwebsocket.html#stateChanged

The various states are here: https://doc.qt.io/qt-5/qabstractsocket.html#SocketState-enum

If that doesn't work the timer will do, I just want to avoid an arbitrary timer if that's possible.